### ISSUE TITLE
patchman -a version KeyError for Visual Studio Code repo

### ISSUE THREAD

**Issue by simonflood** [furlongm/patchman#55]
Hi there,

I've recently found Patchman and I'm trying it with a few Ubuntu 16.04 LTS desktops having installed it on Ubuntu 16.04 LTS Server. Patchman worked fine processing a report from the Server but when I ran "patchman -a" to process the reports from a couple of Desktops it errors on the repo for Visual Studio Code:

--begin--
Repository 33 : Ubuntu 16.04 x86_64 repo at http://packages.microsoft.com/repos/vscode/dists/stable/main/binary-amd64
Found deb repo - http://packages.microsoft.com/repos/vscode/dists/stable/main/binary-amd64/Packages.bz2
Downloading repo info:             100%|#################|Time: 0:00:00
Traceback (most recent call last):  35%|######           |ETA:  0:00:00
  File "/usr/bin/patchman", line 593, in <module>
    main()
  File "/usr/bin/patchman", line 588, in main
    showhelp = process_args(args)
  File "/usr/bin/patchman", line 561, in process_args
    refresh_repos(args.repo, args.force)
  File "/usr/bin/patchman", line 120, in refresh_repos
    repo.refresh(force)
  File "/usr/lib/python2.7/dist-packages/patchman/repos/models.py", line 86, in refresh
    refresh_deb_repo(self)
  File "/usr/lib/python2.7/dist-packages/patchman/repos/utils.py", line 535, in refresh_deb_repo
    packages = extract_deb_packages(data, mirror_url)
  File "/usr/lib/python2.7/dist-packages/patchman/repos/utils.py", line 311, in extract_deb_packages
    fullversion = Version(stanza['version'])
  File "/usr/lib/python2.7/dist-packages/debian/deb822.py", line 245, in __getitem__
    value = self.__dict[key]
KeyError: 'version'
---end---

Server is fully patched with Patchman 1.09-1.

Since this is a test I'm going to try building an Ubuntu 18.04 Server which has an updated python-debian package to see if that behaves better.

Any ideas welcome.

Thanks,
Simon

**Comment by simonflood** [comment]
In case it's relevant perhaps I should note that repo 33 is the first repo to be processed from the Desktops which is different (additional) to the 32 repos in common between the Server and Desktops.

HTH,
Simon

**Comment by simonflood** [comment]
Ubuntu 18.04 is a bust as installing Patchman and running "patchman-manage createsuperuser" both display the following warnings:

--begin--
?: (rest_framework.W001) You have specified a default PAGE_SIZE pagination rest_framework setting,without specifying also a DEFAULT_PAGINATION_CLASS.
	HINT: The default for DEFAULT_PAGINATION_CLASS is None. In previous versions this was PageNumberPagination. If you wish to define PAGE_SIZE globally whilst defining pagination_class on a per-view basis you may silence this check.
---end---

and whilst you can log in via the Patchman URL you then hit an error 500 page.

Now reinstalling server back as Ubuntu 16.04 and will try manually updating python-debian Python module to latest available and retrying tests.

**Comment by simonflood** [comment]
The warnings and 500 error could be related to installing Patchman from Ubuntu 16.04 repo as no 18.04 repo available.

**Comment by furlongm** [comment]
I had a few fixes lined up for the django version on Ubuntu 18.04. I'll dig those out first and commit them. IIRC a new version of python-bootstrap3 is required too (in the meantime, you could also install that from pip rather than deb to get the correct version).

**Comment by furlongm** [comment]
I've pushed the easy fixes for bionic. However, there are still some things to fix on bionic, like celery integration, which I'm still working on. Once that is done, I will create a deb and update the instructions for 18.04. Installing deb on 16.04 is still the best supported method for now.

Regarding the visual studio repo, can you let me know the repo format on the machines, e.g. in `/etc/apt/sources.list` or in a file in `/etc/apt/sources.list.d`?

**Comment by simonflood** [comment]
Thanks for your prompt attention. By modifying /usr/lib/python2.7/dist-packages/patchman/repos/utils.py I was able to proceed beyond the Visual Studio Code repo though I'm now seeing some other oddness which I'll report as a separate issue.

I'll also submit a new issue re Ubuntu 18.04 so we can keep this thread about the original repo issue.

**Comment by simonflood** [comment]
Since your workaround works and is included in 1.0.12 and my other issues are covered by other tickets I think this issue can now be closed. Thanks.
